### ISSUE TITLE
Duplicated notification emails

### ISSUE THREAD

**Issue by dmitrizzle** [roast-cms/analog-cafe-server#253]
If I use two or more images from the same author other than me, they will get the same number of emails as number of images used in the post.

Only one notification per author per publication should be sent out when an image(s) is/are used.

**Comment by ice5050** [comment]
Please test again if it is fixed now

**Comment by dmitrizzle** [comment]
Thank you will test this week 👏

**Comment by dmitrizzle** [comment]
I get the following console output when I start the server on localhost:
```
[nodemon] 1.17.5
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
[nodemon] starting `node --harmony index.js`
server is listening on 8080...
Unhandled rejection MongoError: E11000 duplicate key error collection: analogCafeDev.users index: email_1 dup key: { : "rob.j.davie@gmail.com" }
    at Function.MongoError.create (/Users/dmitri/Documents/Analog.Cafe/api/node_modules/mongoose/node_modules/mongodb-core/lib/error.js:31:11)
    at /Users/dmitri/Documents/Analog.Cafe/api/node_modules/mongoose/node_modules/mongodb-core/lib/connection/pool.js:497:72
    at authenticateStragglers (/Users/dmitri/Documents/Analog.Cafe/api/node_modules/mongoose/node_modules/mongodb-core/lib/connection/pool.js:443:16)
    at Connection.messageHandler (/Users/dmitri/Documents/Analog.Cafe/api/node_modules/mongoose/node_modules/mongodb-core/lib/connection/pool.js:477:5)
    at Socket.<anonymous> (/Users/dmitri/Documents/Analog.Cafe/api/node_modules/mongoose/node_modules/mongodb-core/lib/connection/connection.js:331:22)
    at Socket.emit (events.js:182:13)
    at Socket.EventEmitter.emit (domain.js:442:20)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at Socket.Readable.push (_stream_readable.js:219:10)
    at TCP.onread (net.js:638:20)
```
The server works, but the error persists whenever I restart the server.


When I publish an article, I get no response from node server and the following error:
```
(node:29495) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'id' of undefined
    at self.map.img (/Users/dmitri/Documents/Analog.Cafe/api/helpers/submission.js:260:64)
    at Array.map (<anonymous>)
    at submission.content.raw.document.nodes.filter.map.map.map.filter (/Users/dmitri/Documents/Analog.Cafe/api/helpers/submission.js:260:42)
    at Array.filter (<anonymous>)
    at publish (/Users/dmitri/Documents/Analog.Cafe/api/helpers/submission.js:260:6)
(node:29495) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:29495) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
The article does get published, the email to the main author does get sent. No emails to the secondary authors (image authors who didn't write the article but whose images are part of the submission) are sent at all.

***
Please review those issues.

**Comment by dmitrizzle** [comment]
I fixed the second (fatal) error in #281 - the promise was not resolving and thus no emails were sent.

The first error is related to MongoDB structure and repeat emails, which isn't part of this issue, so I will re-open it separately.
